### ISSUE TITLE
Models missing possibility of multivariate target variable

### ISSUE THREAD

**Issue by max6457** [Nixtla/neuralforecast#396]
I wanted to suggst a new feature: having speperate input and output arrays for the models instead of the dataframe. This would give more possibilities in data preprocessing. Thanks, this is an amasing repository!

**Comment by kdgutier** [comment]
Hi @max6457,

Is there a particular data preprocessing that is not possible to feed
Combining `NeuralForecast.fit` `static_df` and `df` inputs and the `futr_exog_list`, `hist_exog_list`, and `stat_exog_list` inputs of the models?

Here is a usage example of exogenous variables and documentation of the classes:
- https://nixtla.github.io/neuralforecast/examples/exogenous_variables.html
- https://nixtla.github.io/neuralforecast/core.html
- https://nixtla.github.io/neuralforecast/models.nbeatsx.html

Let us know if you find the solution you are looking for.

**Comment by max6457** [comment]
Thanks a lot for your answer. I already used this additional options for other projects.
Now I have the problem, that my input and output arrays have different smoothing... (so it's the same data but smoothed on different ways)
I would need something like: model.train(x_train, y_train) <- fro example how it can be done in tensorflow

**Comment by kdgutier** [comment]
> Thanks a lot for your answer. I already used this additional options for other projects. Now I have the problem, that my input and output arrays have different smoothing... (so it's the same data but smoothed on different ways) I would need something like: model.train(x_train, y_train) <- fro example how it can be done in tensorflow

The PyTorch dataset that feeds the networks are already distinguishing exogenous and target variable.

By "different smoothing" you mean different frequencies of the exogenous and target variable?
If different exogenous frequencies is what you look for,  the NHITS model has built-in multi-rate data sampling capabilities that can be activated with the `n_pool_kernel_size` argument: https://nixtla.github.io/neuralforecast/models.nhits.html

**Comment by max6457** [comment]
Okay, I unterstand that. I will more further look into what you send me and look how much this can help me.
With different smoothing I mean:
input_array = [ [12.5,13,14.5] [13,14.5,13.75] [14.5,13.75, 14] ]
output array = [ [-1,0,1] [-1,1,0.5] [1,0.25, 0.5] ]
This is an example with what I mean. So like for every input there is a specific output -> which means that I can not pass it in an DataFrame...

**Comment by kdgutier** [comment]
Is there a time dependency on your problem, or is it a pure multivariate regression problem?

If you are not modeling output vector interactions, a work-around could be to create the following dataframe:
x1       x2    x3      y
12.5   13  14.5  -1
12.5   13  14.5   0
12.5   13  14.5   1
.
.
.
14.5 13.75 14   0.25
14.5 13.75 14   0.5

Hope it helps

**Comment by max6457** [comment]
Thanks for this idea. I will try it...
But it is a pure multivariate regression problem. 
The easiest way would be an implementation to train the model on seperate input and target arrays as I mentioned before...

**Comment by kdgutier** [comment]
For the moment we don't have multivariate time series regression.
Although I believe that hacking the dataframe might give you something reasonable.

If your problem is "classic" tabular data, random forecast and gradient boosting might be better suited than networks.

**Comment by max6457** [comment]
Thanks! I will be very happy when a new feature to this topic will release :)
